metric-collector/service-readiness/kubeflow-readiness.py

```diff
@@ -0,0 +1,140 @@
+
+def metric_update(args, google_open_id_connect_token):
+  resp = requests.request(
+    METHOD, args.url,
+    headers={'Authorization': 'Bearer {}'.format(
+      google_open_id_connect_token)})
+  if resp.status_code == 200:
+    KUBEFLOW_AVAILABILITY.set(1)
+  else:
+    KUBEFLOW_AVAILABILITY.set(0)
+
+def main(unparsed_args=None):
+  parser = argparse.ArgumentParser(
+    description="Output signal of kubeflow service readiness.")
+
+  parser.add_argument(
+    "--url",
+    default="",
+    type=str,
+    help="kubeflow IAP-protected url")
+  parser.add_argument(
+    "--client_id",
+    default="",
+    type=str,
+    help="Service account json credential file")
+
+  args = parser.parse_args(args=unparsed_args)
+
+  if args.url == "":
+    sleep(2000)
+    return
+
+  # Figure out what environment we're running in and get some preliminary
+  # information about the service account.
+  bootstrap_credentials, _ = google.auth.default(
+    scopes=[IAM_SCOPE])
+  if isinstance(bootstrap_credentials,
+                google.oauth2.credentials.Credentials):
+    raise Exception('make_iap_request is only supported for service '
+                    'accounts.')
+
+  # For service account's using the Compute Engine metadata service,
+  # service_account_email isn't available until refresh is called.
+  bootstrap_credentials.refresh(Request())
+
+  signer_email = bootstrap_credentials.service_account_email
+  if isinstance(bootstrap_credentials,
+                google.auth.compute_engine.credentials.Credentials):
+    # Since the Compute Engine metadata service doesn't expose the service
+    # account key, we use the IAM signBlob API to sign instead.
+    # In order for this to work:
+    #
+    # 1. Your VM needs the https://www.googleapis.com/auth/iam scope.
+    #    You can specify this specific scope when creating a VM
+    #    through the API or gcloud. When using Cloud Console,
+    #    you'll need to specify the "full access to all Cloud APIs"
+    #    scope. A VM's scopes can only be specified at creation time.
+    #
+    # 2. The VM's default service account needs the "Service Account Actor"
+    #    role. This can be found under the "Project" category in Cloud
+    #    Console, or roles/iam.serviceAccountActor in gcloud.
+    signer = google.auth.iam.Signer(
+      Request(), bootstrap_credentials, signer_email)
+  else:
+    # A Signer object can sign a JWT using the service account's key.
+    signer = bootstrap_credentials.signer
+
+  # Construct OAuth 2.0 service account credentials using the signer
+  # and email acquired from the bootstrap credentials.
+  service_account_credentials = google.oauth2.service_account.Credentials(
+    signer, signer_email, token_uri=OAUTH_TOKEN_URI, additional_claims={
+      'target_audience': args.client_id
+    })
+
+  token_refresh_time = 0
+  while True:
+    if time() > token_refresh_time:
```

Comment:
Either in this PR or in a follow on PR; could we emit Kubernetes events as well indicating the status of the endpoint.

Reply:
For all service / deployment, will fetch with prometheus and store in stackdriver by default.
Will do in separate PR.